fhd_core/setup_metadata/fhd_struct_init_layout.pro

```diff
@@ -0,0 +1,268 @@
+function fhd_struct_init_layout, ant_table_header, ant_table_data, $
+    array_center = array_center, coordinate_frame = coordinate_frame, $
+    gst0 = gst0, earth_degpd = earth_degpd, ref_date = ref_date, time_system = time_system, $
+    dut1 = dut1, diff_utc = diff_utc, nleap_sec = nleap_sec, $
+    pol_type = pol_type, n_pol_cal_params = n_pol_cal_params, n_antenna = n_antenna, $
+    antenna_names = antenna_names, antenna_numbers = antenna_numbers, antenna_coords = antenna_coords, $
+    mount_type = mount_type, axis_offset = axis_offset, pola = pola, pola_orientation = pola_orientation, $
+    pola_cal_params = pola_cal_params, polb = polb, polb_orientation = polb_orientation, polb_cal_params = polb_cal_params
+    
+    
+  if n_elements(ant_table_header) ne 0 then begin
+    ;; first get values out of header
+    keyword_list = list()
+    for si = 0, n_elements(ant_table_header)-1 do begin
+      keyword_list.add, strlowcase(strtrim(strmid(ant_table_header[si], 0, 8)))
+    endfor
+    if keyword_list.count('arrayx') gt 0 and keyword_list.count('arrayy') gt 0 and keyword_list.count('arrayz') gt 0 then $
+      array_center = [sxpar(ant_table_header,'arrayx'), sxpar(ant_table_header,'arrayy'), sxpar(ant_table_header,'arrayz')]
+      
+    if keyword_list.count('frame') gt 0 then coordinate_frame = sxpar(ant_table_header,'frame')
```

Comment:
What order of precedence do you want here? Right now the order is found in header -> user supplied -> default value. That makes sense if the user is supplying improved default values, but the values found in the header will be used no matter what if they exist. However, a user might expect their values to be used instead of the header values if they have taken the trouble to supply them in their wrapper or on the command line.